plugins/modules/collaborator_information.py

```diff
@@ -0,0 +1,353 @@
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.permissions.push:
+    description: Will return true if push rights are given to collaborator. Read, clone, and push to repository.
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.permissions.pull:
+    description: Will return true if pull rights are given to collaborator. Read and clone repository.
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.permissions.triage:
+    description: Will return true if triage rights are given to collaborator.
+                 Triage role can request reviews on pull requests (PRs), mark issues and PRs as duplicates, and add or remove milestones on issues and PRs.
+                 NO WRITE ACCESS.
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.site_admin:
+    description: Will return true if collaborator is a site admin.
+                 This permission gives the collaborator the ability to manage users, organizations, and repositories.
+    type: bool
+    returned: only if at least one collaborator is contained within repository
+
+collaborators['<ORG NAME>/<REPO NAME>'].<INDEX>.type:
+    description: This will return what type of collaborator the user is.
+    type: str
+    returned: only if at least one collaborator is contained within repository
+'''
+
+
+def add_collaborators(g, repos, to_add):
+    for repo in repos:
+        r = g.get_repo(repo)
+        collaborator_list = []
+        collaborators = r.get_collaborators(affiliation="direct")
+        for collaborator in collaborators:
+            collaborator_list.append(collaborator.login)
+
+        for p in to_add:
+            r.add_to_collaborators(p, permission=to_add[p])
+
+
+def check_permissions(g, repos, user_to_check):
+    status = True
+    for repo in repos:
+        r = g.get_repo(repo)
+        for user in user_to_check.items():
+            if(r.get_collaborator_permission(user[0]) != user[1]):
+                status = False
+    return status
+
+
+def del_collaborators(g, repos, to_remove):
+    for repo in repos:
+        r = g.get_repo(repo)
+        collaborators = r.get_collaborators(affiliation="direct")
+        for collaborator in collaborators:
+            if(collaborator.login in to_remove):
+                r.remove_from_collaborators(collaborator.login)
+
+
+def change_collaborator_permissions(g, repos, to_change):
+    for repo in repos:
+        r = g.get_repo(repo)
+        collaborator_list = []
+        collaborators = r.get_collaborators(affiliation="direct")
+        for collaborator in collaborators:
+            collaborator_list.append(collaborator.login)
```

Comment:
Use a set instead of a list.  Using a list results in line 247 having to do a linear search each time it is called.